parlai/tasks/vqa_v1/agents.py

```diff
@@ -44,6 +47,24 @@ def _path(opt):
     return data_path, annotation_path, image_path
 
 
+class MasterLoader(Thread):
```

Comment:
so if this stuff works, we move it to other tasks as well, or what?

Reply:
Yes, Alex and I spoke about extending this to a new parent teacher that would allow a subclass to make use of this